src/main/java/com/adyen/builders/terminal/TerminalAPIRequestBuilder.java

```diff
@@ -0,0 +1,425 @@
+        this.cardReaderPowerOffRequest = cardReaderPowerOffRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withReconciliationRequest(ReconciliationRequest reconciliationRequest) {
+        this.messageClass = MessageClassType.SERVICE;
+        this.messageCategory = MessageCategoryType.RECONCILIATION;
+        this.reconciliationRequest = reconciliationRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withReversalRequest(ReversalRequest reversalRequest) {
+        this.messageClass = MessageClassType.SERVICE;
+        this.messageCategory = MessageCategoryType.REVERSAL;
+        this.reversalRequest = reversalRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withSoundRequest(SoundRequest soundRequest) {
+        this.messageClass = MessageClassType.DEVICE;
+        this.messageCategory = MessageCategoryType.SOUND;
+        this.soundRequest = soundRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withStoredValueRequest(StoredValueRequest storedValueRequest) {
+        this.messageClass = MessageClassType.SERVICE;
+        this.messageCategory = MessageCategoryType.STORED_VALUE;
+        this.storedValueRequest = storedValueRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withTransactionStatusRequest(TransactionStatusRequest transactionStatusRequest) {
+        this.messageClass = MessageClassType.SERVICE;
+        this.messageCategory = MessageCategoryType.TRANSACTION_STATUS;
+        this.transactionStatusRequest = transactionStatusRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withTransmitRequest(TransmitRequest transmitRequest) {
+        this.messageClass = MessageClassType.DEVICE;
+        this.messageCategory = MessageCategoryType.TRANSMIT;
+        this.transmitRequest = transmitRequest;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withSecurityTrailer(ContentInformation securityTrailer) {
+        this.securityTrailer = securityTrailer;
+        return this;
+    }
+
+    public TerminalAPIRequest build() {
+        MessageHeader messageHeader = new MessageHeader();
+        messageHeader.setProtocolVersion(PROTOCOL_VERSION);
+        messageHeader.setMessageType(MessageType.REQUEST);
+        messageHeader.setMessageClass(messageClass);
+        messageHeader.setMessageCategory(messageCategory);
+        messageHeader.setSaleID(saleId);
+        messageHeader.setServiceID(serviceId);
+        messageHeader.setPOIID(poiId);
+
+        SaleToPOIRequest saleToPOIRequest = new SaleToPOIRequest();
+        saleToPOIRequest.setMessageHeader(messageHeader);
+        saleToPOIRequest.setSecurityTrailer(securityTrailer);
+
+        if (messageCategory != null) {
+            switch (messageCategory) {
+                case PAYMENT:
+                    saleToPOIRequest.setPaymentRequest(paymentRequest);
```

Comment:
null checks ? same for all below ?

Reply:
Talked about last week, it might be null on misuse of the builder, but the request response will explicit show that a requested was missing